packages/core/src/amazonq/activation.ts

```diff
@@ -17,6 +17,86 @@ import { listCodeWhispererCommandsWalkthrough } from '../codewhisperer/ui/status
 import { Commands } from '../shared/vscode/commands2'
 import { focusAmazonQPanel, focusAmazonQPanelKeybinding } from '../codewhispererChat/commands/registerCommands'
 import { TryChatCodeLensProvider, tryChatCodeLensCommand } from '../codewhispererChat/editor/codelens'
+import { Auth } from '../auth'
+import { customLearnMoreUri } from '../codewhisperer/models/constants'
+import { openUrl } from '../shared/utilities/vsCodeUtils'
+import { AuthUtil } from '../codewhisperer'
+import { ConnectionStateChangeEvent } from '../auth/auth'
+
+/**
+ * Display a notification to user for Log In.
+ *
+ * Authentication Notification is displayed when:
+ * - The user closes the Amazon Q chat panel, and
+ * - The user has not performed any authentication action.
+ *
+ * Error Notification is displayed when:
+ * - The user closes the Amazon Q chat panel, and
+ * - The user attempts an authentication action but is not logged in.
+ *
+ * @param {Event} isAmazonQVisibleEvent - Event indicating the visibility status of the Amazon Q chat.
+ * @param {Event} onDidUpdateConnectionEvent - Event indicating the authentication connection update.
+ */
+async function showAuthNotification(
+    isAmazonQVisibleEvent: vscode.Event<boolean>,
```

Comment:
We should use the `on***` variable naming for events to keep things consistent.

Possible solution:
```suggestion
    onAmazonQChatVisibility: vscode.Event<boolean>,
```